src/trading/tradingSdk.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  LimitOrderAdvancedSettings,
+  LimitTradeParameters,
+  QuoteAndPost,
+  SwapAdvancedSettings,
+  TradeParameters,
+  TraderParameters,
+} from './types'
+import { postSwapOrder, postSwapOrderFromQuote } from './postSwapOrder'
+import { postLimitOrder } from './postLimitOrder'
+import { getQuoteWithSigner } from './getQuote'
+import { postSellNativeCurrencyTrade } from './postSellNativeCurrencyTrade'
+import { getSigner, swapParamsToLimitOrderParams } from './utils'
+import { getPreSignTransaction } from './getPreSignTransaction'
+
+export class TradingSdk {
+  constructor(public readonly traderParams: TraderParameters) {}
+
+  async getQuote(params: TradeParameters, advancedSettings?: SwapAdvancedSettings): Promise<QuoteAndPost> {
+    const quoteResults = await getQuoteWithSigner(this.mergeParams(params), advancedSettings)
+
+    return {
+      quoteResults: quoteResults.result,
+      postSwapOrderFromQuote: () => postSwapOrderFromQuote(quoteResults),
+    }
+  }
+
+  async postSwapOrder(params: TradeParameters, advancedSettings?: SwapAdvancedSettings): Promise<string> {
+    return postSwapOrder(this.mergeParams(params), advancedSettings)
+  }
+
+  async postLimitOrder(params: LimitTradeParameters, advancedSettings?: LimitOrderAdvancedSettings): Promise<string> {
+    return postLimitOrder(this.mergeParams(params), advancedSettings)
+  }
+
+  async postSellNativeCurrencyTrade(
+    params: TradeParameters,
+    advancedSettings?: SwapAdvancedSettings
+  ): Promise<ReturnType<typeof postSellNativeCurrencyTrade>> {
+    const quoteResults = await getQuoteWithSigner(this.mergeParams(params), advancedSettings)
+
+    const { tradeParameters, quoteResponse, amountsAndCosts } = quoteResults.result
+    return postSellNativeCurrencyTrade(
+      quoteResults.orderBookApi,
+      quoteResults.result.signer,
+      quoteResults.result.appDataInfo,
+      // Quote response response always has an id
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      swapParamsToLimitOrderParams(tradeParameters, quoteResponse.id!, amountsAndCosts)
+    )
+  }
+
+  async getPreSignTransaction(params: { orderId: string; account: string }): ReturnType<typeof getPreSignTransaction> {
```

Comment:
Any reason why it's not in the same pattern? Something like: postPresignOrder.
Then this fn would post the order to the backend and send the tx to the wallet/signer.

Reply:
Because this function returns only a transaction, we can't create an order on behalf of smart-contract wallet.
The maximum we can do is to build a transaction so the smart-contract wallet could only execute it.

Reply:
Yes we can, and that's actually what we do with preSign https://github.com/cowprotocol/cowswap/blob/4b9331cbfab92fc113043c4e204e280b55d09f62/apps/cowswap-frontend/src/legacy/utils/trade.ts#L226-L253

![image](https://github.com/user-attachments/assets/611692ed-e913-4abb-b5d8-524c89c8b609)

Reply:
Sorry, I meant we can't execute a transaction to sign the order on behalf of the smart-contract wallet.
I've added docs for this case: https://github.com/cowprotocol/cow-sdk/pull/213/commits/31b149869162df2303fa82841ab99e9ad0d806d9
I hope now it's clear

Reply:
As you can see, we need `orderId` for `getPreSignTransaction()`, so we have to created order first anyway

Reply:
As it was clarified in a call, now I got it.
All good.